server/src/BoardSync.ts

```diff
@@ -5,19 +5,39 @@ import { BoardState } from "common/shared";
 import { broadcast } from "./broadcast";
 
 export class BoardSync {
+    static PLACING_COOLDOWN_PERIOD: number = 500; // ms
     private socketsWithTruth: Array<Socket> = [];
     private broadcastUpdateBoard: broadcast["updateBoard"];
+    private lastPlacingTS: number = new Date().getTime();
 
     constructor(broadcastUpdateBoard: broadcast["updateBoard"]) {
         this.broadcastUpdateBoard = broadcastUpdateBoard;
         setInterval(async () => {
             const socket = this.pickRandomSourceOfTruth();
             if (!socket) return;
+
             const boardState = await this.getBoardFromClient(socket);
+            const now = new Date().getTime();
+            if (now - this.lastPlacingTS < BoardSync.PLACING_COOLDOWN_PERIOD) {
+                console.log(
+                    "BoardSync: skipping update at",
+                    now,
+                    ", last placing event: ",
+                    this.lastPlacingTS,
+                    "diff:",
+                    now - this.lastPlacingTS
+                );
+                // skip when sync is less than 200ms after the previous placing event
+                return;
+            }
```

Comment:
What if instead of having a cooldown period, we declared `now` (currently line 20) above the await for the state. After receiving the state, we then check if any placements occurred while waiting for the state (after `now`), and if so, cancel the state update.

This would reduce the number of skipped states. Consider the case where a place occurs at t=-100ms, broadcasted by the server at t=-75ms, and then the server requests state from the source of truth at t=-50ms, receiving the response at t=0ms. In such a case, we would skip the state update even though the order of events is valid. The only time the order isn't valid is if the place is broadcast after the request for state, which we can check directly by the method above.